diff --git a/youtube_dl/extractor/generic.py b/youtube_dl/extractor/generic.py
index 77e217460..9bd979ff4 100644
--- a/youtube_dl/extractor/generic.py
+++ b/youtube_dl/extractor/generic.py
@@ -3168,19 +3168,22 @@ class GenericIE(InfoExtractor):
         entries = self._parse_html5_media_entries(url, webpage, video_id, m3u8_id='hls')
         if entries:
             if len(entries) == 1:
-                entries[0].update({
+                entry = entries[0]
+                entry.update({
                     'id': video_id,
                     'title': video_title,
                 })
+                self._sort_formats(entry['formats'])
+                return entry
             else:
                 for num, entry in enumerate(entries, start=1):
                     entry.update({
                         'id': '%s-%s' % (video_id, num),
                         'title': '%s (%d)' % (video_title, num),
                     })
-            for entry in entries:
-                self._sort_formats(entry['formats'])
-            return self.playlist_result(entries, video_id, video_title)
+                for entry in entries:
+                    self._sort_formats(entry['formats'])
+                return self.playlist_result(entries, video_id, video_title)
 
         jwplayer_data = self._find_jwplayer_data(
             webpage, video_id, transform_source=js_to_json)
