diff --git a/youtube_dl/extractor/twitter.py b/youtube_dl/extractor/twitter.py
index cebb6238c..0ee97f434 100644
--- a/youtube_dl/extractor/twitter.py
+++ b/youtube_dl/extractor/twitter.py
@@ -175,11 +175,17 @@ class TwitterCardIE(TwitterBaseIE):
                 u, video_id, headers={'Referer': 'https://twitter.com/'})
 
             iframe_url = self._html_search_regex(
-                r'<iframe[^>]+src="((?:https?:)?//(?:www\.youtube\.com/embed/[^"]+|(?:www\.)?vine\.co/v/\w+/card))"',
+                r'<iframe[^>]+src="((?:https?:)?//(?:www\.)?vine\.co/v/\w+/card)"',
                 webpage, 'video iframe', default=None)
             if iframe_url:
                 return self.url_result(iframe_url)
 
+            iframe_url = self._html_search_regex(
+                r'<iframe[^>]+src="((?:https?:)?//www\.youtube\.com/embed/[^"]+)"',
+                webpage, 'video iframe', default=None)
+            if iframe_url:
+                raise ExtractorError('Embedded YouTube video is not accepted', expected=True)
+
             config = self._parse_json(self._html_search_regex(
                 r'data-(?:player-)?config="([^"]+)"', webpage,
                 'data player config', default='{}'),
